src/libp2p/service.rs

```diff
@@ -14,30 +14,22 @@ use crate::message::SignedMessage;
 use crate::{blocks::GossipBlock, rpc_api::net_api::NetInfoResult};
 use crate::{chain::ChainStore, utils::encoding::from_slice_with_fallback};
 use ahash::{HashMap, HashSet};
-use anyhow::Context as _;
 use cid::Cid;
 use flume::Sender;
-use futures::{
-    channel::oneshot::Sender as OneShotSender, future::Either, select, stream::StreamExt,
-};
+use futures::{channel::oneshot::Sender as OneShotSender, select, stream::StreamExt};
```

Comment:
nit, I know the code was already like this
```suggestion
use futures::{channel::oneshot, select, stream::StreamExt as _};
```
Then use `oneshot::Sender`